### PR TITLE
Move docs to readthedocs

### PR DESCRIPTION
**Pull request by sravel**
First template to use sphinx and readthedocs.
see #251

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+autodocsumm>=0.1.13
+autodoc>=0.3
+sphinx>=3.1.2
+sphinx_automodapi>=0.12
+sphinxcontrib-napoleon>=0.7
```
Comment (anxuae) [inline docs/requirements.txt:5]: Remove it: now included in Sphinx as `sphinx.ext.napoleon` extension

File: docs/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+sphinxcontrib-autoprogram
+BioPython
```
Comment (anxuae) [inline docs/requirements.txt:7]: Useless

File: docs/make.bat [hunk 3]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (anxuae) [inline docs/make.bat:1]: Delete this file: never been generated on Windows

File: docs/source/README.rst [hunk 4]
```diff
@@ -0,0 +1,421 @@
+|Pibooth| |BeerPay|
```
Comment (anxuae) [inline docs/source/README.rst:1]: this file shall be removed and split is different sub ReST files

File: docs/source/conf.py [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = ['sphinx.ext.intersphinx',
+              'sphinx.ext.viewcode',
+              'sphinx.ext.autodoc',
+              'sphinx.ext.autosectionlabel',
+              'autodocsumm',
+              # 'sphinx_automodapi.automodapi',
```
Comment (anxuae) [inline docs/source/conf.py:35]: Remove

File: docs/source/conf.py [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+              'sphinx.ext.napoleon',
+              # 'sphinxcontrib.autoprogram'
```
Comment (anxuae) [inline docs/source/conf.py:37]: Remove

File: docs/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,7 @@
+autodocsumm>=0.1.13
+autodoc>=0.3
+sphinx>=3.1.2
+sphinx_automodapi>=0.12
```
Comment (anxuae) [inline docs/requirements.txt:4]: Remove useless

File: docs/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+sphinxcontrib-napoleon>=0.7
+sphinxcontrib-autoprogram
```
Comment (anxuae) [inline docs/requirements.txt:6]: Remove useless

File: docs/source/index.rst [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+   ./dev.rst
+
+.. toctree::
+   :caption: Plugin
+   :name: Plugin
+   :maxdepth: 3
+
+   ./plugin.rst
+
+.. toctree::
```
Comment (anxuae) [inline docs/source/index.rst:40]: Delete because finally never used

File: docs/source/conf.py [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+html_theme = 'sphinx_rtd_theme'
+
+# Theme options are theme-specific and customize the look and feel of a theme
+# further.  For a list of options available for each theme, see the
+# documentation.
+html_theme_options = {
+        'canonical_url'             : '',
+        'analytics_id'              : 'UA-XXXXXXXXXXXXX-1',  # Provided by Google in your dashboard
```
Comment (anxuae) [inline docs/source/conf.py:152]: delete because not used

File: docs/source/conf.py [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+# The short X.Y version.
+version = "2.0.3"
+
+if os.environ.get('READTHEDOCS') == 'True':
+    # Because Read The Docs modifies conf.py, versioneer gives a "dirty"
+    # version like "5.10.0+0.g28674b1.dirty" that is cleaned here.
+    version = version.partition('+0.g')[0]
+
+# The full version, including alpha/beta/rc tags.
+release = "2.0.3"
```
Comment (anxuae) [inline docs/source/conf.py:100]: import pibooth, then use pibooth.__version__ to get a dynamic update of the version

File: docs/source/conf.py [hunk 12]
```diff
@@ -0,0 +1,340 @@
...
+
+latex_authors = ''''''
+
+# import
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = "2.0.3"
```
Comment (anxuae) [inline docs/source/conf.py:92]: import pibooth , then pibooth.__version__

File: docs/source/conf.py [hunk 13]
```diff
@@ -0,0 +1,340 @@
...
+
+if os.environ.get('READTHEDOCS') == 'True':
```
Comment (anxuae) [inline docs/source/conf.py:94]: Stange, I've already made some doc on readthedoc, and never seen that

File: docs/source/conf.py [hunk 14]
```diff
@@ -0,0 +1,340 @@
...
+# If false, no module index is generated.
+# texinfo_domain_indices = True
+
+# How to display URL addresses: 'footnote', 'no', or 'inline'.
+# texinfo_show_urls = 'footnote'
+
+# If true, do not generate a @detailmenu in the "Top" node's menu.
+# texinfo_no_detailmenu = False
+
+def setup(app):
```
Comment (anxuae) [inline docs/source/conf.py:338]: Why you are not using the `html_style` ?
The original css from readthedoc theme seems ok to me.

File: docs/source/conf.py [hunk 15]
```diff
@@ -0,0 +1,340 @@
...
+# html_theme_path = sphinx_bootstrap_theme.get_html_theme_path()
+
+# The name for this set of Sphinx documents.  If None, it defaults to
+# "<project> v<release> documentation".
+html_title = "Pibooth"
+
+# A shorter title for the navigation bar.  Default is the same as html_title.
+html_short_title = "Pibooth"
+
+# The name of an image file (relative to this directory) to place at the top
```
Comment (anxuae) [inline docs/source/conf.py:175]: better to keep all image in the "images" directory

File: docs/source/conf.py [hunk 16]
```diff
@@ -0,0 +1,340 @@
...
+autosectionlabel_maxdepth = 2
+autosummary_generate = True
+autosummary_generate_overwrite = True
+# numpydoc settings
+numpydoc_show_class_members = True
+numpydoc_show_inherited_class_members = True
+numpydoc_attributes_as_param_list = False
+numpydoc_class_members_toctree = False
+
+# Napoleon settings
```
Comment (anxuae) [inline docs/source/conf.py:50]: Napoleon is not necessary, we don't use lumpy enough and pibooth don't use the Google style docstrings

File: docs/source/conf.py [hunk 17]
```diff
@@ -0,0 +1,340 @@
...
+              'sphinx.ext.napoleon',
+              # 'sphinxcontrib.autoprogram'
+              ]
+
+autosectionlabel_prefix_document = True
+autosectionlabel_maxdepth = 2
+autosummary_generate = True
+autosummary_generate_overwrite = True
+# numpydoc settings
+numpydoc_show_class_members = True
```
Comment (anxuae) [inline docs/source/conf.py:45]: All info related to numpy are necessary

**Review by anxuae — CHANGES_REQUESTED** [review]
Thanks for the PR.
I have two global comments:

 - it seems that the config is coming from an other project and there is lots of parameters and extension which are useless. The best is to restart with clean setup using the command `sphinx-quickstart` (I can do it if you don't have the time)
- keep a clean directory structure: all images inside the same `images` directory which is separated from the sources, alls `_statics` shall be outside the sources

**Comment by sravel** [comment]
yes I made this first PR to see with you how to organize directories, how split files.... If you are agree I completely change docs structures to have beautiful readthedocs!!

File: docs/source/conf.py (hunk 14 above)
Comment (sravel) [reply docs/source/conf.py]: I use old project as template, bad idea...

File: docs/source/conf.py (hunk 12 above)
Comment (sravel) [reply docs/source/conf.py]: OK

File: docs/source/conf.py (hunk 11 above)
Comment (sravel) [reply docs/source/conf.py]: OK

**Comment by sravel** [comment]
I build from sphinx-quickstart.
need to update README but i don't know how to shorten

**Pull request merged**